docs/Extended_dry_run.rst

```diff
@@ -0,0 +1,731 @@
+.. _extended_dry_run:
+
+Extended dry run
+================
+
+Using ``--extended-dry-run`` or ``-x`` (supported since EasyBuild v2.4.0), a detailed overview of the build and install
```

Comment:
You mean release notes?

I can, but they still need to be fleshed out for the upcoming v2.4.0.

I can get them started and only include an entry for https://github.com/hpcugent/easybuild-framework/pull/1388 though, so I will, and include a reference here.